### PR TITLE
POC: redact sensitive storage account info from debug logger

### PR DESCRIPTION
**Pull request by yugangw-msft**
@lmazuel, please let me know whether the patch here is sustainable, otherwise, we need to expose a hook from the pipeline for client applications to redact in anyway they prefer; or any other options you would suggest.
 
CC @Juliehzl

- [x] The PR has modified HISTORY.rst describing any customer-facing, functional changes. Note that this does not include changes only to help content. (see [Modifying change log](https://github.com/Azure/azure-cli/tree/master/doc/authoring_command_modules#modify-change-log)).

- [x] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

### CODE REVIEW

File: src/azure-cli/azure/cli/command_modules/storage/_validators.py [hunk 1]
```diff
@@ -40,7 +40,7 @@ def _query_account_key(cli_ctx, account_name):
 
 def _query_account_rg(cli_ctx, account_name):
     """Query the storage account's resource group, which the mgmt sdk requires."""
-    scf = get_mgmt_service_client(cli_ctx, ResourceType.MGMT_STORAGE)
```
Comment (Juliehzl) [inline src/azure-cli/azure/cli/command_modules/storage/_validators.py]: Why not apply `storage_client_factory` for [_query_account_key](https://github.com/Azure/azure-cli/blob/2bc61d23b447585415b6d10611d36bc4c8c4f3a6/src/azure-cli/azure/cli/command_modules/storage/_validators.py#L29)? It looks `_query_account_rg` is used for listing storage accounts and `_query_account_key` is for listing keys.

Comment (yugangw-msft) [reply src/azure-cli/azure/cli/command_modules/storage/_validators.py]: Good question. Sorry this change is not very relevant,  as I wanted to have one less way to create the storage client. That is it. I can undo this if you like

File: src/azure-cli/azure/cli/command_modules/storage/_client_factory.py [hunk 2]
```diff
@@ -138,6 +138,15 @@ def cf_sa(cli_ctx, _):
     return storage_client_factory(cli_ctx).storage_accounts
 
 
+def cf_sa_for_keys(cli_ctx, _):
+    from knack.log import get_logger
+    logger = get_logger(__name__)
+    logger.debug('Disable HTTP logging to avoid having storage keys in debug logs')
+    client = storage_client_factory(cli_ctx)
+    client.config.enable_http_logger = False
```
Comment (jiasli) [inline src/azure-cli/azure/cli/command_modules/storage/_client_factory.py]: ✔Tested with `storage account keys list -n {} --debug`.

File: src/azure-cli/azure/cli/command_modules/storage/_validators.py [hunk 3]
```diff
@@ -32,6 +32,8 @@ def _query_account_key(cli_ctx, account_name):
     t_storage_account_keys = get_sdk(
         cli_ctx, ResourceType.MGMT_STORAGE, 'models.storage_account_keys#StorageAccountKeys')
 
+    scf.config.enable_http_logger = False
```
Comment (jiasli) [inline src/azure-cli/azure/cli/command_modules/storage/_validators.py]: ✔Tested with `storage blob copy start --source-uri {} --account-name {} -b {} -c {} --debug`

**Pull request merged**